### PR TITLE
Add install and update scripts, don't add powerline function if user doesn't install it

### PR DESCRIPTION
**Pull request by ShaneSmiskol**
- Probably could add a function to .bashrc to trigger `update.py` in the future
- `install.py` is to be ran on first clone, however `update.py` runs `install.py` so you could probably combine the two. Minimal error checking right now
- Detect if powerline is installed; if it's not, use a different `.bashrc` file without that function so it doesn't freak the user out with errors upon ssh connection after installing this repo
- Use spaces in the `.bashrc` files
  - `.bashrc_powerline` is to be identical to `.bashrc`, just without the top powerline function

Method of installing this repo changes from cloning the files directly to `/home` to `/home/comma-dotfiles` so that an installation script can copy and replace old files in `/home` for easy updates. Automatically replaces the files in `/home` that are defined here: https://github.com/ShaneSmiskol/comma-dotfiles/blob/master/install.py#L6

Let me know what you think of these ideas! I like the idea of a 'workbench for developers', that's the feeling I got from the project.

### CODE REVIEW

**Comment by AskAlice** [comment]
Could you write this with a dont-repeat-yourself philosophy with the bashrc? Just tell bash to source if powerline-shell is installed

**Comment by AskAlice** [comment]
Secondly, this is great, I will merge this. Just would be great to see that happen

File: .bashrc_powerline [hunk 1]
```diff
@@ -0,0 +1,20 @@
+cd /data/openpilot
+function _update_ps1() {
+      PS1=$(powerline-shell $?)
+  }
+
+  if [[ $TERM != linux && ! $PROMPT_COMMAND =~ _update_ps1 ]]; then
+        PROMPT_COMMAND="_update_ps1; $PROMPT_COMMAND"
+  fi
+
+function pandaflash() {
```
Comment (AskAlice) [inline .bashrc_powerline:10]: DRY

**Comment by ShaneSmiskol** [comment]
Awesome, glad to hear you like it. I actually have always used Windows my entire life so I don't know too much about linux-specific stuff like its scripting. I'll look into it though

**Comment by ShaneSmiskol** [comment]
@AskAlice How does that look now?

**Comment by AskAlice** [comment]
hmm possible, sounds a bit overcomplicated. In bash you could probably test to make sure the output of `pip show powerline-shell` is nonzero, if it is nonzero, then powerline-shell is installed

**Comment by AskAlice** [comment]
oh wait, you did that.. hold up let me review this more lol

**Comment by ShaneSmiskol** [comment]
Yeah, just found a command online that seems to check if the command exists, and is true if it is! It worked in my quick test

**Comment by AskAlice** [comment]
you all set on merging this?

**Comment by ShaneSmiskol** [comment]
Let me shorten the install file first, then you can merge

**Comment by AskAlice** [comment]
kk

**Comment by AskAlice** [comment]
oooh im tempted to `pip install thefuck` and add that to the bashrc

**Comment by ShaneSmiskol** [comment]
Ooh, I haven't seen that before. That's great haha. Latest commit is working, I'll just unify update and install now. Is `updatedotfiles` a good name? I wonder how hard it would be to allow commands like `dotfiles update`, or similar

**Comment by ShaneSmiskol** [comment]
Yeah definitely make sure you squash merge this PR lol, should be good if you think the name for the new command is okay

**Comment by ShaneSmiskol** [comment]
I got this syntax working, which I think straight up looks cooler, but requires more to be typed for the same to be accomplished. Do you want me to replace the current commands with this? `dotfiles pandaflash`, `dotfiles update`, etc...

Screenshot_625

**Pull request merged**